sdk/core/azure-core/tests/async_tests/test_streaming_async.py

```diff
@@ -203,3 +222,126 @@ async def test_compress_compressed_header(http_request):
+        assert decoded == "test"
+
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("http_request", HTTP_REQUESTS)
+async def test_compress_plain_header_offline(port, http_request):
+    # expect plain text
+    client = AsyncPipelineClient("")
+    async with client:
+        request = http_request(method="GET", url="http://localhost:{}/streams/plain_header".format(port))
+        pipeline_response = await client._pipeline.run(request,stream=True)
+        response = pipeline_response.http_response
+        data = response.stream_download(client._pipeline,decompress=False)
+        content = b""
+        async for d in data:
+            content += d
+        decoded = content.decode("utf-8")
+        assert decoded == "test" 
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("http_request", HTTP_REQUESTS)
+async def test_decompress_compressed_no_header_offline(port, http_request):
+    # expect compressed text
+    client = AsyncPipelineClient("")
+    async with client:
+        request = http_request(method='GET',url="http://localhost:{}/streams/compressed_no_header".format(port))
+        pipeline_response = await client._pipeline.run(request,stream=True)
+        response = pipeline_response.http_response
+        data = response.stream_download(client._pipeline,decompress=True)
+        content = b""
+        async for d in data:
+            content += d
+        with pytest.raises(UnicodeDecodeError):
+            content.decode("utf-8")
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("http_request", HTTP_REQUESTS)
+async def test_compress_compressed_header_offline(port, http_request):
+    # expect compressed text
+    client = AsyncPipelineClient("")
+    async with client:
+        request = http_request(method='GET',url="http://localhost:{}/streams/compressed_header".format(port))
+        pipeline_response = await client._pipeline.run(request,stream=True)
+        response = pipeline_response.http_response
+        data = response.stream_download(client._pipeline,decompress=False)
+        content = b""
+        async for d in data:
+            content += d
+        with pytest.raises(UnicodeDecodeError):
+            content.decode("utf-8")         
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize("http_request", HTTP_REQUESTS)
+async def test_decompress_plain_header_offline(port, http_request):
+    # expect error
+    client = AsyncPipelineClient("")
+    async with client:
+        request = http_request(method="GET", url="http://localhost:{}/streams/compressed".format(port))
+        pipeline_response = await client._pipeline.run(request,stream=True)
+        response = pipeline_response.http_response
+        print(f"this is the pipeline respone {response}")
+        data = response.stream_download(client._pipeline,decompress=True)
+        try:
+            content = b""
+            async for d in data:
+                content += d
+            assert False
+        except (zlib.error, DecodeError):
```

Comment:
you can also use `pytest.raises` here, and expect one of these errors to show up